### PR TITLE
Complete ign to gz migration

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. Not ready yet, as mission tests against latest `lrauv-application` are failing -.-

### CODE REVIEW

**Comment by hidmic** [comment]
> mission tests against latest lrauv-application are failing -.-

Aha! They are failing on `main` too, see https://github.com/osrf/lrauv/pull/248. I'll solve that there, then come back.

**Comment by hidmic** [comment]
@caguero This is getting hairy. Tests that use the LRAUV controller fail seemingly at random, or at least I cannot see the pattern just yet.

**Comment by hidmic** [comment]
Finally passing. @adityapande-1995 @arjo129 PTAL!

File: docker/base/Dockerfile [hunk 1]
```diff
@@ -32,27 +30,62 @@ RUN echo 'Etc/UTC' > /etc/timezone && \
...
+        sudo \
+        tzdata \
+        vim \
+ && apt-get clean
 
-RUN sudo /bin/sh -c 'echo "deb [arch=amd64,arm64] http://repo.ros2.org/ubuntu/main `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2-latest.list' \
- && curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | sudo apt-key add - \
- && sudo apt-get update \
- && sudo apt-get install -y python3-vcstool python3-colcon-common-extensions
+# Install Ignition build tools
```
Comment (adityapande-1995) [inline docker/base/Dockerfile:58]: Minor comment : Ignition -> Gazebo

**Review by adityapande-1995 — COMMENTED** [review]
Hmm, are we not running the mission tests anymore ? I see the files were deleted.

File: docker/base/Dockerfile [hunk 2]
```diff
@@ -32,27 +30,62 @@ RUN echo 'Etc/UTC' > /etc/timezone && \
 
 RUN apt-get update \
  && apt-get install -y \
-        build-essential \
+        cmake \
+        cppcheck \
         curl \
+        gdb \
```
Comment (adityapande-1995) [inline docker/base/Dockerfile:36]: Nit : Do we need gdb, and libbluetooth for the docker ?

**Review by adityapande-1995 — APPROVED** [review]

File: docker/base/Dockerfile (hunk 2 above)
Comment (hidmic) [reply docker/base/Dockerfile:36]: Ahh, my mistake. I'll open up a follow-up PR.

**Pull request merged**